packages/core/test/unit/query-generator/add-constraint-query.test.ts

```diff
@@ -0,0 +1,420 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { Deferrable, Op } from '@sequelize/core';
+import { createSequelizeInstance, expectsql, sequelize } from '../../support';
+
+const dialect = sequelize.dialect;
+const notSupportedError = new Error(`Add constraint queries are not supported by ${dialect.name} dialect`);
+const checkNotSupportedError = new Error(`Check constraints are not supported by ${dialect.name} dialect`);
+const defaultNotSupportedError = new Error(`Default constraints are not supported by ${dialect.name} dialect`);
+const deferrableNotSupportedError = new Error(`Deferrable constraints are not supported by ${dialect.name} dialect`);
+const onUpdateNotSupportedError = new Error(`Foreign key constraint with onUpdate is not supported by ${dialect.name} dialect`);
+
+describe('QueryGenerator#addConstraintQuery', () => {
+  const queryGenerator = sequelize.getQueryInterface().queryGenerator;
+
+  it('throws an error if invalid type', () => {
+    // @ts-expect-error -- We're testing invalid options
+    expectsql(() => queryGenerator.addConstraintQuery('myTable', { type: 'miss-typed', fields: ['otherId'] }), {
+      default: new Error(`Constraint type miss-typed is not supported by ${dialect.name} dialect`),
+      sqlite: notSupportedError,
+    });
+  });
+
+  it('calls _getConstraintSnippet function', () => {
+    const addConstraintQuerySpy = sinon.stub(queryGenerator, 'addConstraintQuery');
+    queryGenerator.addConstraintQuery('myTable', { type: 'UNIQUE', fields: ['myColumn'] });
+    expect(queryGenerator.addConstraintQuery).to.have.been.calledWith('myTable', { type: 'UNIQUE', fields: ['myColumn'] });
+    addConstraintQuerySpy.restore();
+  });
+
+  describe('CHECK constraints', () => {
+    it('generates a constraint snippet for a check constraint with a name', () => {
```

Comment:
```suggestion
    it('generates a query that adds a check constraint with a name', () => {
```

etc, because they're not snippets anymore :)

Reply:
Done